### PR TITLE
docs: add concepts for project definition and deploy command

### PR DESCRIPTION
**Pull request by RonWang**
Solve https://github.com/alexa-labs/ask-cli/issues/41 by adding the docs

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by RonWang** [comment]
Revision 2 raised @breedloj please check!

File: docs/concepts/Alexa-Skill-Project-Definition.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+Please check the `skillMetadata` field in the example below for its representation in project config.
+
+
+## Skill Code
+The `SkillCode` component manages the source code for each region. CLI will be in charge of the building process for the code, and passing the code built result to the infrastructure deployer. We support code management for all the regions that [Alexa supports](https://developer.amazon.com/en-US/docs/alexa/custom-skills/host-a-custom-skill-as-an-aws-lambda-function.html#select-the-optimal-region-for-your-aws-lambda-function).
+
+Please check the `code` field in the example below for its representation in project config.
+
+## Skill Infrastructure
+The `SkillInfrastructure` represents the configuration on how to deploy skill's code, and tracks the deployment status for continuous deployment. This deployer platform is designed to cope with the variety of serverless frameworks, with necessary interfaces including `userConfig` and `deployState`. For more details, please check [how to create an Alexa skill deployer](TODO).
```
Comment (pbheemag) [inline docs/concepts/Alexa-Skill-Project-Definition.md]: TODO needs to be addressed?

File: docs/concepts/Deploy-Command.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
+# Deploy Command
+
+`askx deploy` deploys all the Alexa skill project resources by following the settings from the project config file (i.e. ask-resources.json). It will deploy each [resource component](./Alexa-Skill-Project-Definition.md) from the local file(s) to the target endpoint sequentially. The deploy flow for each component has been optimized to be executed only when new changes exist in the local (see [usage of lastDeployHash]()). Users of deploy command can focus more on the content within each component, and only control the deployment when it's necessary.
```
Comment (pbheemag) [inline docs/concepts/Deploy-Command.md]: link needs to be updated for `resource component`, `usage of lastDeployHash` and `here` (about ask-resources config file).

Comment (RonWang) [reply docs/concepts/Deploy-Command.md]: The resource component one works only when you combine them together, as it's a relative path. I will add the usage of lastDeployHash and link it to the sentence.

**Comment by RonWang** [comment]
Revision 3 done addressing @pbheemag's comment by adding the usage of lastDeployHash.

**Pull request merged**